seqr/views/utils/dataset_utils.py

```diff
@@ -111,39 +111,41 @@ def _find_or_create_missing_sample_records(
         samples,
         projects,
         user,
-        sample_ids,
+        sample_project_tuples,
+        get_individual_sample_key,
         raise_no_match_error=False,
         raise_unmatched_error_template=None,
         create_active=False,
         get_individual_sample_lookup=_get_individual_sample_lookup,
-        get_individual_sample_key=None,
         **kwargs
 ):
     samples = list(samples)
-    remaining_sample_ids = set(sample_ids) - {sample.sample_id for sample in samples}
+    remaining_sample_keys = set(sample_project_tuples) - {(sample.sample_id, sample.project_name) for sample in samples}
```

Comment:
We annotate a project_name field for the existing samples. It looks cleaner than the `sample.individual.family.project.name`, and it embedded the logic of using `None` for non-RNA and no-project info data.

Reply:
Adding a helper function argument for getting the remaining sample keys is a great idea. I will do that.